src/Parser/Core/Modules/MitigationCheck.js

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import SPELLS from 'common/SPELLS/index';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber, formatPercentage } from 'common/format';
+
+import SpellLink from 'common/SpellLink';
+import Enemies from 'Parser/Core/Modules/Enemies';
+
+import {findByBossId} from 'Raids/index';
+import ExpandableStatisticBox from 'Main/ExpandableStatisticBox';
+
+const mitigationBuffs = {'Paladin' : [SPELLS.SHIELD_OF_THE_RIGHTEOUS_BUFF.id,
+                                      SPELLS.ARDENT_DEFENDER.id,
+                                      SPELLS.GUARDIAN_OF_ANCIENT_KINGS.id],
+                          'Monk' : [SPELLS.IRONSKIN_BREW_BUFF.id,
+                                    SPELLS.FORTIFYING_BREW_BRM_BUFF.id,
+                                    SPELLS.ZEN_MEDITATION.id,
+                                    SPELLS.DAMPEN_HARM_TALENT.id,
+                                    ],
+                          'Warrior': [SPELLS.SHIELD_BLOCK_BUFF.id,
+                                      SPELLS.IGNORE_PAIN.id,
+                                      SPELLS.LAST_STAND.id,
+                                      SPELLS.SHIELD_WALL.id,
+                                      SPELLS.SPELL_REFLECTION.id],
+                          'DemonHunter': [SPELLS.DEMON_SPIKES_BUFF.id,
+                                          SPELLS.METAMORPHOSIS_TANK.id],
+                          'Druid': [SPELLS.IRONFUR.id,
+                                    SPELLS.FRENZIED_REGENERATION.id,
+                                    SPELLS.BARKSKIN.id,
+                                    SPELLS.SURVIVAL_INSTINCTS.id],
+                          'DeathKnight': [SPELLS.BONE_SHIELD.id,
+                                          SPELLS.BLOOD_SHIELD.id,
+                                          SPELLS.ANTI_MAGIC_SHELL.id,
+                                          SPELLS.VAMPIRIC_BLOOD.id,
+                                          SPELLS.ICEBOUND_FORTITUDE.id,
+                                          SPELLS.DANCING_RUNE_WEAPON_BUFF.id,
+                                          SPELLS.TOMBSTONE_TALENT.id]};
+
+const mitigationDebuffs = {'Paladin' : [],
+                            'Monk' : [SPELLS.BREATH_OF_FIRE_DEBUFF.id],
+                            'Warrior': [SPELLS.DEMORALIZING_SHOUT.id],
+                            'DemonHunter': [SPELLS.FIERY_BRAND_DEBUFF.id],
+                            'Druid': [],
+                            'DeathKnight': []};
+
+
+class MitigationCheck extends Analyzer{
+  static checks;
```

Comment:
These properties should not be marked as static, they're being used as regular class properties.

A static property is for something that couldn't different between instances (e.g. is the same over multiple fights). If something can be different per fight (or instance of the module), it should be a regular class property, which is just `checks;` without `static`.

Reply:
Giving properties a default value of at least `null` would be preferred. Without doing this the value of vars is `undefined`, which is generally a value that should be left for things that are truly undefined. A class property is essentially a definition, and if something doesn't have a default value `null` is a great indicator of that.